src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -1,27 +1,181 @@
-@reexport module ITensorChainRules
+module ITensorChainRules
 
 using ChainRulesCore
 using ITensors
 
+include("zygoterules.jl")
+
 function ChainRulesCore.rrule(::typeof(getindex), x::ITensor, I...)
   y = getindex(x, I...)
   function getindex_pullback(ȳ)
-    x̄ = ITensor(inds(x))
+    # TODO: add definition `ITensor(::Tuple{}) = ITensor()`
+    # to ITensors.jl so no splatting is needed here.
+    x̄ = ITensor(inds(x)...)
     x̄[I...] = ȳ
     Ī = broadcast(_ -> NoTangent(), I)
     return (NoTangent(), x̄, Ī...)
   end
   return y, getindex_pullback
 end
 
+# Specialized version in order to avoid call to `setindex!`
+# within the pullback, should be better for taking higher order
+# derivatives in Zygote.
+function ChainRulesCore.rrule(::typeof(getindex), x::ITensor)
+  y = x[]
+  function getindex_pullback(ȳ)
+    x̄ = ITensor(ȳ)
+    return (NoTangent(), x̄)
+  end
+  return y, getindex_pullback
+end
+
+function setinds_pullback(ȳ, x, a...)
+  x̄ = ITensors.setinds(ȳ, inds(x))
+  ā = broadcast(_ -> NoTangent(), a)
+  return (NoTangent(), x̄, ā...)
+end
+
+for fname in (
```

Comment:
Same question here

Reply:
Yes agreed that I didn't take into account that the Index ordering could be different in the reverse pass. I will think about the best approach for that.